### PR TITLE
fix (kubernetes-maven-plugin) : Resource Validation is done before resources are generated by `k8s:resource` (#2008)

### LINKED ISSUE #2008: Resource Validation is done before resources are generated by `k8s:resource`
### Describe the bug

I observed this while working on https://github.com/eclipse/jkube/issues/1934

On a fresh build project, `k8s:resource` does not perform any resource validation:
```
$ mvn clean install k8s:resource
```
However, if you've already run `k8s:resource` before, you'd see validation on the second run. This behavior is due to validation method invoked before resource files are actually written:

https://github.com/eclipse/jkube/blob/174d1897655b8af31637d79728169768d2cf8004/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojo.java#L216-L217

Gradle Plugins don't seem to affected by this issue. They're correctly invoking validation after resource generation:
https://github.com/eclipse/jkube/blob/174d1897655b8af31637d79728169768d2cf8004/gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/KubernetesResourceTask.java#L91-L92

### Eclipse JKube version

1.10.0

### Component

Kubernetes Maven Plugin

### Apache Maven version

3.8.5

### Gradle version

None

### Steps to reproduce

Run mvn clean install k8s:resource , no validation would be performed

### Expected behavior

Validation should be performed after resource generation. Lines 216 and 127 should be swapped in ResourceMojo.java

### Runtime

Kubernetes (vanilla)

### Kubernetes API Server version

1.25.3

### Environment

Linux

### Eclipse JKube Logs

_No response_

### Sample Reproducer Project

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description
Fixes #2008

Resource validation should be done after resources are written.

Signed-off-by: Rohan Kumar <rohaan@redhat.com>

## Type of change

 - [X] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] My code follows the style guidelines of this project
 - [X] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [X] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [X] I have performed a self-review of my code
 - [X] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [X] I have added tests that prove my fix is effective or that my feature works
 - [X] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ Usage:
 * Fix #1935: `oc:remote-dev` goal / `ocRemoteDev` task have wrong log prefixes
 * Fix #1966: Old reference to fmp in documentation
 * Fix #1974: Remove unused methods in KubernetesResourceUtil
+* Fix #2008: Resource Validation is done before resources are generated by `k8s:resource`
```
Comment (manusa) [inline CHANGELOG.md:31]: I'd rephrase this to
```suggestion
* Fix #2008: resources validated after their generation by `k8s:resource`
```

to explain the solution provided instead of the problem described in the issue.

**Review by manusa — CHANGES_REQUESTED** [review]
nit: just some changes to the changelog entry

**Review by sunix — APPROVED** [review]

**Review by manusa — APPROVED** [review]
